### PR TITLE
AUT-1257: remove client reg dependancy

### PR DESCRIPTION
**Pull request by gn-GDS**
## What?

In the di-authentication-frontend in landing-controller.ts we rely on the start response to populate the client in the information in the session. We should populate this client information with the claims we receive from the Authentication request from Orchestration. 

Note - Do not remove the logic in the landing-controller.ts as this behaviour is controlled by a feature flag and we still rely on this behaviour in Production.

## Why?

Decouple client registry as part of Auth/Orch split

## Related PRs
[AUT-1248: Decrypt JWE and verify JWT signature] (https://github.com/alphagov/di-authentication-frontend/pull/1096)

### CODE REVIEW

**Review by DTaylorGDS — APPROVED** [review]
Looks good to me. The only thing that really jumps out (and not particularly in scope of this PR, since the same criticism can be levelled at the controller we used as a base (the landing controller)) is the functional complexity of the handler/controller. Even just breaking some stuff out into smaller sub-functions might help a bit in terms of readability (caveat, to save time, I just pasted your existing function code into ChatGPT and asked it to break it into smaller functions, so take the exact snippet below with a pinch of salt):

```
export function authorizeGet(
  authService: AuthorizeServiceInterface = authorizeService(),
  cookieService: CookieConsentServiceInterface = cookieConsentService(),
  kmsService: KmsDecryptionServiceInterface = new KmsDecryptionService(),
  jwtService: JwtServiceInterface = new JwtService()
): ExpressRouteFunc {
  return async function (req: Request, res: Response) {
    try {
      const claims = await processAuthRequest(req, kmsService, jwtService);
      const startAuthResponse = await startAuthorization(res.locals);

      updateSession(req, claims, startAuthResponse);
      handleCookieConsent(req, res, cookieService, startAuthResponse);

      const redirectPath = determineRedirectPath(req, startAuthResponse);
      return res.redirect(redirectPath);
    } catch (error) {
      handleAuthorizationError(error, res);
    }
  };
}

async function processAuthRequest(
  req: Request,
  kmsService: KmsDecryptionServiceInterface,
  jwtService: JwtServiceInterface
) {
  // ... process auth request and return claims
}

async function startAuthorization(locals: any) {
  // ... start authorization and return response
}

function updateSession(req: Request, claims: any, startAuthResponse: any) {
  // ... update session based on claims and response
}

function handleCookieConsent(
  req: Request,
  res: Response,
  cookieService: CookieConsentServiceInterface,
  startAuthResponse: any
) {
  // ... handle cookie consent logic
}

function determineRedirectPath(
  req: Request,
  startAuthResponse: any
): string {
  // ... determine and return the redirect path
}

function handleAuthorizationError(error: any, res: Response) {
  // ... handle authorization errors and send appropriate response
}
```

**Review by BeckaL — COMMENTED** [review]
Some minor comments, looks good to me otherwise

File: src/components/authorize/jwt-service.ts [hunk 1]
```diff
@@ -11,17 +11,22 @@ export class JwtService implements JwtServiceInterface {
     this.publicKey = publicKey;
   }
 
-  async getPayloadWithValidation(jwt: string): Promise<any> {
+  async getPayloadWithValidation(jwt: string): Promise<Claims> {
+    let claims: any;
```
[the remark this answers is not in the document]
Comment (gn-GDS) [reply src/components/authorize/jwt-service.ts:15]: @BeckaL  I guess its technically a "jose.JWTPayload", it doesn't become a Claims object until its been through validation.

[the remark this answers is not in the document]
Comment (BeckaL) [reply src/components/authorize/jwt-service.ts]: Is it possible to put that in? Having an `any` in production code seems like a bit of a smell to me unless absolutely necessary

**Review by BeckaL — APPROVED** [review]
Nice, LGTM

**Pull request merged**